crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -32,6 +32,73 @@ def role_constraints
     end
   end
 
+  def shutdown_openstack_at_nodes
+    db_nodes = []
+    NodeObject.find("state:crowbar_upgrade").each do |node|
```

Comment:
Hm, ok, my comment was not relevant. So, what could go wrong? I do not think anything should go wrong, because in this stage, recipe only shuts down the services (non-DB). If they are  already stopped, that's fine (we have such case anyway in case of HA cluster)